Bio/AlignIO/MafIO.py

```
@@ -0,0 +1,581 @@
+        try:
+            line = next(handle)
+        except StopIteration:
+            line = ""
+
+        if in_a_bundle:
+            if line.startswith("s"):
+                # add a SeqRecord to the bundle
+                line_split = line.strip().split()
+
+                if len(line_split) != 7:
+                    raise ValueError("Error parsing alignment - 's' line must have 7 fields")
+
+                # convert MAF-style +/- strand to biopython-type 1/-1
+                if line_split[4] == "+":
+                    strand = 1
+                elif line_split[4] == "-":
+                    strand = -1
+                else:
+                    strand = 1
+
+                # s (literal), src (ID), start, size, strand, srcSize, text (sequence)
+                anno = {"start": int(line_split[2]),
+                        "size": int(line_split[3]),
+                        "strand": strand,
+                        "srcSize": int(line_split[5])}
+
+                sequence = line_split[6]
+
+                # interpret a dot/period to mean same the first sequence
+                if "." in sequence:
+                    if not records:
+                        raise ValueError("Found dot/period in first sequence of alignment")
+
+                    ref = str(records[0].seq)
+                    new = []
+
+                    for (s, r) in zip(sequence, ref):
+                        new.append(r if s == "." else s)
+
+                    sequence = "".join(new)
+
+                records.append(SeqRecord(Seq(sequence, alphabet),
+                               id=line_split[1],
+                               name=line_split[1],
+                               description="",
+                               annotations=anno))
+            elif line.startswith("e") or \
+                 line.startswith("i") or \
+                 line.startswith("q"):
+                # not implemented
```

http://genome.ucsc.edu/FAQ/FAQformat.html#format5 says

> Some MAF files may contain other optional line types:
> 
> an "i" line containing information about what is in the aligned species DNA before and after the immediately preceding "s" line
> an "e" line containing information about the size of the gap between the alignments that span the current block
> a "q" line indicating the quality of each aligned base for the species

As they are optional an error seems too much. Silently ignoring them (as now) seems reasonable.